vendor/python/astexport.py

```diff
@@ -87,18 +87,18 @@ def visit_field_Num_n(self, val):
         if isinstance(val, int):
```

Comment:
ah, okay – this should probably just be

```python
def visit_field_Num_n(val):
    return str(val)
```

this value won't actually be used in the printer (which as with the JS printer will only look at the raw string value), so anything JSON-serializable that will work for the AST comparison will be fine

Reply:
You lose a bit of information in doing this, does it hurt to keep around the source type?

Reply:
i don't think you do, actually, relative to what's in the AST:

```
In [1]: str(1)
Out[1]: '1'

In [2]: str(1.)
Out[2]: '1.0'

In [3]: str(0xDEADBEEF)
Out[3]: '3735928559'

In [4]: str(1_000_001)
Out[4]: '1000001'

In [5]: str(5j)
Out[5]: '5j'

In [6]: str(3-4j)
Out[6]: '(3-4j)'
```

there might be some edge cases on float rounding, but you'd already encounter them with the above